sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,340 @@
+    '_map_chain_code_columns_to_coordinates',
+]
+
+
+def _freeze(obj):
+    """
+    Create a hashable representation of a dict or list.
+    Otherwise returns the original input.
+    """
+    if isinstance(obj, dict):
+        return tuple((k, _freeze(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return tuple(_freeze(elem) for elem in obj)
+    return obj
+
+
+def _get_unit_attributes():
+    """
+    Returns the attributes from the HEK that are used to parse the values in the HEK table.
+    """
+    unit_properties_filename = get_pkg_data_filename('data/unit_properties.json',
+                                                     package='sunpy.net.hek')
+    with open(unit_properties_filename) as unit_file:
+        unit_properties = json.load(unit_file)
+    return unit_properties["attributes"]
+
+
+def _get_coord_attributes():
+    coord_properties_filename = get_pkg_data_filename('data/coord_properties.json',
+                                                      package='sunpy.net.hek')
+    with open(coord_properties_filename) as coord_file:
+        coord_properties = json.load(coord_file)
+    return coord_properties["attributes"]
+
+
+def _map_columns_to_times(table):
+    """
+    Represent time columns in an HEK response as astropy Time objects.
+    """
+    # All time columns from https://www.lmsal.com/hek/VOEvent_Spec.html
+    time_keys = ['event_endtime', 'event_starttime', 'event_peaktime']
+    for tkey in time_keys:
+        if tkey in table.colnames:
+            table[tkey] = [
+                (parse_time(time, format='iso') if time else np.ma.masked) for time in table[tkey]
+            ]
+
+
+def _map_columns_to_quantities(table):
+    """
+    For columns in an HEK response which have associated units, combine
+    data and unit columns into masked columns with units.
+    """
+    attributes = _get_unit_attributes() + _get_coord_attributes()
+    missing_values = {
+        'integer': -999,
+        'long': -999,
+        'float': np.nan,
+    }
+    dtype_aliases = {
+        'string': 'str',
+        'integer': 'int',
+    }
+    for attr in attributes:
+        if (name := attr["name"]) not in table.colnames:
+            continue
+        if attr.get("is_unit_prop", False):
+            continue
+        if attr.get("is_chaincode", False):
+            continue
+        mask = np.array([r is None for r in table[name]])
+        # NOTE: Fill with missing values explicitly because None cannot be cast
+        # to a all dtypes.
+        data = np.where(mask, missing_values.get(attr['type'], None), table[name])
+        if unit_prop := attr.get("unit_prop", False):
+            if unit_prop not in table.colnames:
+                log.debug(f"Missing unit property {unit_prop} for {name}. Using event_coordunit.")
+                unit_prop = "event_coordunit"
+            units = np.array([_parse_unit(_u) if _u is not None else '' for _u in table[unit_prop]])
```

Comment:
This is epically pedantic, but it would be better to use `u_` here to avoid a collision with `u` rather than `_u` which would imply private `u`.
```suggestion
            units = np.array([_parse_unit(u_) if u_ is not None else '' for u_ in table[unit_prop]])
```

Reply:
What are we if not especially pedantic around here.